### PR TITLE
New FW receive flow

### LINKED ISSUE #7698: Adjust Suite for receiving ActionCancelled from Trezor in receive flow
With new receive flow on FW, user can cancel address confirmation if the address in Trezor and Suite are different. 

Right now, when Trezor sends `ActionCancelled` to Suite, Suite will freeze.

More info here: https://github.com/trezor/trezor-firmware/pull/2843#issuecomment-1441622398

### PR DESCRIPTION
**Pull request by marekrjpolak**
## Description
Minimal change to fix issue #7698. Now all possible modals are explicitly closed when `TrezorConnect.getAddress` is unsuccessful, not only the `address` one.

More optional changes (refactoring mostly) are in followup PR #7964.

## Related issue
Resolve #7698

### CODE REVIEW

File: packages/suite/src/actions/wallet/receiveActions.ts [hunk 1]
```diff
@@ -148,15 +148,7 @@ export const showAddress =
...
-            // we are blocking close event in the modal reducer.
-            // On failed action we need to close it manually
-            const currentModal = getState().modal;
-            if (
-                currentModal.context === '@modal/context-user' &&
-                currentModal.payload.type === 'address'
-            ) {
-                dispatch(modalActions.onCancel());
-            }
+            dispatch(modalActions.onCancel());
```
Comment (matejkriz) [inline packages/suite/src/actions/wallet/receiveActions.ts:151]: does it mean that it closes any modal that is open at the moment? for example coinjoin critical phase?

Comment (marekrjpolak) [reply packages/suite/src/actions/wallet/receiveActions.ts:151]: Hmm. If it's possible to initiate receive address flow from Suite, show the address (any of the two possible screens) on device display and in that moment critical phase starts, then yes. But is it even possible to do a CJ 'takeover' of the device while in some other flow?

Comment (Hannsek) [reply packages/suite/src/actions/wallet/receiveActions.ts:151]: When the critical phase starts, on Trezor, there are screen from which is impossible tu send the message, no?

Comment (matejkriz) [reply packages/suite/src/actions/wallet/receiveActions.ts:151]: @marekrjpolak  yes, for some cases "takeover" is possible, and CJ is just an example see https://github.com/trezor/trezor-suite/pull/7656 (not sure if there is currently any other [override](https://github.com/trezor/trezor-suite/pull/7628/commits/0c0c1bbe7e0cac60324efe2667f314624ea6e76a))

Comment (marekrjpolak) [reply packages/suite/src/actions/wallet/receiveActions.ts:151]: I've tried and (at least in this case) it's ok. Critical phase first cancels the receive flow, kills the modal and only after that critical phase modal is shown.

Comment (matejkriz) [reply packages/suite/src/actions/wallet/receiveActions.ts:151]: sounds good 👍

**Review by matejkriz — APPROVED** [review]
👍

**Pull request merged**